### PR TITLE
Multi attachments

### PR DESCRIPTION
**Pull request by ar4s**
TODO:
- [x] add test
- [x] custom migration
- [x] squash commits

### CODE REVIEW

File: src/ralph/lib/transitions/migrations/0006_auto_20170719_1209.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def rewrite_rewind(apps, schema_editor):
+    TransitionsHistory = apps.get_model('transitions', 'TransitionsHistory')
+    for th in TransitionsHistory.objects.all():
+        for i, attachment in enumerate(th.attachments.all()):
+            if i == 0:
+                th.attachment = attachment
+            else:
+                print('Attachment {} for transition history ({}) will be lost.'.format(attachment.file.url, th.transition_name))  # noqa
```
Comment (romcheg) [inline src/ralph/lib/transitions/migrations/0006_auto_20170719_1209.py]: To be a good scout it's possible to zip/tar/winrar attachments, if there are more than one. However, I assume nobody's gonna use this rewind, so I'll live it up to your consideration :)

File: src/ralph/back_office/models.py [hunk 2]
```diff
@@ -528,7 +530,7 @@ def _generate_report(cls, name, request, instances, language):
         form_fields={
             'report_language': {
                 'field': forms.ModelChoiceField(
-                    label=_('Report language'),
+                    label=_('Report language (release)'),
```
Comment (romcheg) [inline src/ralph/back_office/models.py]: Release report language?

File: src/ralph/back_office/models.py [hunk 3]
```diff
@@ -549,7 +551,7 @@ def release_report(cls, instances, request, **kwargs):
         form_fields={
             'report_language': {
                 'field': forms.ModelChoiceField(
-                    label=_('Report language'),
+                    label=_('Report language (return)'),
```
Comment (romcheg) [inline src/ralph/back_office/models.py]: ditto

File: src/ralph/back_office/models.py [hunk 4]
```diff
@@ -570,7 +572,7 @@ def return_report(cls, instances, request, **kwargs):
         form_fields={
             'report_language': {
                 'field': forms.ModelChoiceField(
-                    label=_('Report language'),
+                    label=_('Report language (loan)'),
```
Comment (romcheg) [inline src/ralph/back_office/models.py]: ditto

File: src/ralph/admin/templatetags/ralph_tags.py [hunk 5]
```diff
@@ -23,8 +23,8 @@
 )
 def download_attachment(context):
     return {
-        'attachment': context.request.session.pop(
-            'attachment_to_download', None
+        'attachments': context.request.session.pop(
+            'attachment_to_download', []
```
Comment (mkurek) [inline src/ralph/admin/templatetags/ralph_tags.py]: `attachments_to_download`?

File: src/ralph/lib/transitions/migrations/0006_auto_20170719_1209.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    TransitionsHistory = apps.get_model('transitions', 'TransitionsHistory')
+    for th in TransitionsHistory.objects.all():
+        if th.attachment:
+            th.attachments.add(th.attachment)
+
+
+def rewrite_rewind(apps, schema_editor):
+    attachments_backup = '/tmp/attachments_backup.zip'
+    TransitionsHistory = apps.get_model('transitions', 'TransitionsHistory')
+    attachemnts = ZipFile(attachments_backup, 'w')
```
Comment (mkurek) [inline src/ralph/lib/transitions/migrations/0006_auto_20170719_1209.py]: typo here

**Review by mkurek — CHANGES_REQUESTED** [review]

**Review by mkurek — APPROVED** [review]

**Pull request merged**